File: live-test/kotlin/src/main/kotlin/demo/AsyncSuper.kt
// Suspend-function super resolution: the suspend modifier must not stop
// AsyncImpl.fetch from resolving to the suspend fun declared in AsyncFetcher.
package demo

interface AsyncFetcher {
    suspend fun fetch(): String
}

class AsyncImpl : AsyncFetcher {
    override suspend fun fetch(): String = "fetched"
}

File: live-test/python/src/async_super.py
# Super-method fixture: the async modifier must not break override resolution —
# AsyncImpl.fetch still resolves to the abstract AsyncFetcher.fetch.
"""Async def override across ABC + impl."""
from abc import ABC, abstractmethod


class AsyncFetcher(ABC):
    @abstractmethod
    async def fetch(self) -> str:
        ...


class AsyncImpl(AsyncFetcher):
    async def fetch(self) -> str:
        return "fetched"
